api/apiclient.go

```diff
@@ -621,26 +664,32 @@ func gorillaDialWebsocket(ctx context.Context, urlStr string, tlsConfig *tls.Con
 }
 
 type resolvedAddress struct {
-	host string
 	ip   string
 	port string
+	url  *url.URL
 }
 
+// addressProvider provides a mechanism for resolving
+// host addresses to IP addresses. It utilises a DNS cache and an IP
+// address resolver to manage and resolve addresses. It maintains
+// separate pools for addresses that need to be resolved and those that
+// have been resolved via the DNS cache.
 type addressProvider struct {
 	dnsCache       DNSCache
 	ipAddrResolver IPAddrResolver
 
 	// A pool of host addresses to be resolved to one or more IP addresses.
```

Comment:
This comment is a bit out of date now? "A pool of host URLs...". Same below.
Do we need the full URL if we are just wanting to get the resolved IP at the end of the day? Wouldn't just having addrPool be the list of host names that need resolving be sufficient?